satpy/readers/__init__.py

```diff
@@ -477,7 +476,10 @@ def find_files_and_readers(start_time=None, end_time=None, base_dir=None,
                 base_dir, reader, sensor, reader_configs, reader_kwargs, fs)
         sensor_supported = sensor_supported or this_sensor_supported
         if loadables:
-            reader_files[reader_instance.name] = list(loadables)
+            if fs is not None and isinstance(fs, fsspec.spec.AbstractFileSystem):
+                reader_files[reader_instance.name] = [FSFile(fn, fs=fs) for fn in loadables]
+            else:
+                reader_files[reader_instance.name] = list(loadables)
```

Comment:
This could be a separate function, for example `_collect_loadables(fs, loadables, reader_instance_name, reader_files)` which would update the `reader_files` list.